scripts/powershell/Verify-And-Merge-PRs.ps1

```diff
@@ -0,0 +1,145 @@
+
+        $data = @{
+            commit_title = ""
+            commit_message = ""
+            sha = $obj.HeadSHA
+            merge_method = "squash"
+        }
+
+        # Get latest Commit of the pull request
+        try 
+        {
+            $response = Invoke-RestMethod -Method Get -Headers $Headers $CommitsAPIUrl
+            $data.commit_message = $response[$response.Count - 1].commit.message # last commit message of the pull request
+            $data.commit_title = "Merge pull request #$($obj.PRNumber) from $($obj.HeadLabel)"
+        }
+        catch 
+        {
+            Write-Error "Invoke-RestMethod [$CommitsAPIUrl] failed with exception:`n$_"
+            exit 1
+        }
+
+        # Merge Pull Request
+        try 
+        {
+            $response = Invoke-RestMethod -Method Put -Headers $Headers $MergeAPIUrl -Body ($data | ConvertTo-Json)
+        }
+        catch 
+        {
+            Write-Output "Could not merge https://github.com/$($obj.RepoOwner)/$($obj.RepoName)/pull/$($obj.PRNumber)"
+            Write-Error "Invoke-RestMethod [$MergeAPIUrl] failed with exception:`n$_"
+            exit 1
+        }
+    }
+}
+
+function Confirm-Mergability()
+{
+    foreach ($obj in $PRObjs)
+    {
+        $APIUrl = "${GitHubAPIUrlBase}$($obj.RepoOwner)/$($obj.RepoName)/pulls/$($obj.PRNumber)"
+        try
+        {
+            $response = Invoke-RestMethod -Headers $Headers $APIUrl
+            if ($response.merged)
+            {
+                [void]$MergedPRs.Add($obj)
+            }
+            elseif ($response.state -eq "closed")
+            {
+                [void]$ClosedPRs.Add($obj)
+            }
+            else 
+            {
+                if ($response.mergeable_state -ne "clean")
+                {
+                    [void]$OpenButDirtyPRs.Add($obj)
+                }
+                elseif ($response.mergeable -and ($response.mergeable_state -eq "clean"))
+                {
+                    $obj | Add-Member -MemberType NoteProperty -Name "HeadSHA" -Value $response.head.sha
+                    $obj | Add-Member -MemberType NoteProperty -Name "HeadLabel" -Value $response.head.label
+                    [void]$OpenAndCleanPRs.Add($obj)
+                }
+                else 
+                {
+                    [void]$UnexpectedState.Add($obj)
+                }
+            }
+        }
+        catch
+        {
+            Write-Error "Invoke-RestMethod ${APIUrl} failed with exception:`n$_"
+            exit 1
+        }
+    }
+}
+function Get-PRDataFromFile()
+{
+    $PRData = Get-Content $PRDataArtifactPath
+    foreach ($line in $PRData)
+    {
+        $PRDetails = $line.Split(';')
+        $PRDataObj = [PSCustomObject]@{
+            RepoOwner = $PRDetails[0]
+            RepoName = $PRDetails[1]
+            PRNumber = $PRDetails[2]
+        }
+        [void]$PRObjs.Add($PRDataObj)
+    }
+}
+
+Get-PRDataFromFile
```

Comment:
I'm not sure I would have created these as functions if you don't plan to reuse them and you are just mutating and consuming global variables. I would just make this a linear script so it is easier to walk through.